### PR TITLE
AS-714: Prevent retries when emails fail to send due to missing keys

### PR DESCRIPTION
**Pull request by MatthewBemis**

### CODE REVIEW

File: src/main/scala/thurloe/notification/NotificationMonitor.scala [hunk 1]
```diff
@@ -121,7 +121,12 @@ class NotificationMonitorActor(val pollInterval: FiniteDuration, pollIntervalJit
 
     lookupShouldNotify(notification) flatMap { shouldNotify =>
       if (shouldNotify) {
-        sendGridDAO.sendNotifications(List(toThurloeNotification(notification))).map(_.headOption)
+        sendGridDAO.sendNotifications(List(toThurloeNotification(notification))).map(_.headOption) recoverWith {
+          case e: KeyNotFoundException =>
+            logger.info(s"Unable to send notification due to missing key: ${e.getMessage}")
+            Future.successful(None)
```
Comment (MatthewBemis) [inline src/main/scala/thurloe/notification/NotificationMonitor.scala:127]: returning None here will have the same effect as the user having turned off notifications: we will ack the pubsub message so we don't attempt to send it again

the lack of the ack is what caused the log spam

File: src/main/scala/thurloe/notification/NotificationMonitor.scala [hunk 2]
```diff
@@ -121,7 +121,12 @@ class NotificationMonitorActor(val pollInterval: FiniteDuration, pollIntervalJit
...
+          case e => throw e
```
Comment (MatthewBemis) [inline src/main/scala/thurloe/notification/NotificationMonitor.scala:128]: i figure it is still worth throwing exceptions if they are for some other reason, but could also be convinced that we should ack them and log the exception at the `warn` level

**Review by davidangb — APPROVED** [review]
:+1: for this fix, what you have here is good and beneficial to merge.

For future thought ... afaict the logic in [HttpSendGridDAO.lookupPreferredEmail ](https://github.com/broadinstitute/thurloe/blob/9bc5589f1fee47f425af10398735f0fda9350e7e/src/main/scala/thurloe/dataaccess/HttpSendGridDAO.scala#L28-L33)is faulty. I think this method is _supposed_ to use `contactEmail` if it exists, else use `email`. But, it looks like that method will throw if `contactEmail` does not exist, and thus never use `email`, potentially resulting in some users not receiving notifications when they should.

**Comment by MatthewBemis** [comment]
@davidangb I think you're right about that. I've filed AS-741 to track it

**Review by ungwudik — APPROVED** [review]

**Pull request merged**